File: lecture_009(Recursion)/l_010_displayreversearray.java
import java.util.Scanner;
public class l_010_displayreversearray {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        displayArrReverse(arr,0);
    }
    public static void input(int[] arr) {
        for(int i=0; i<arr.length; i++) {
            arr[i] = scn.nextInt();
        }
    }
    public static void displayArrReverse(int[] arr, int idx) {
        if(idx==arr.length) {
            return;
        }
        displayArrReverse(arr,idx+1);
        System.out.println(arr[idx]);
    }
}

File: lecture_009(Recursion)/l_010_displayreversearray.cpp
#include<iostream>
#include<vector>
using namespace std;

void input(vector<int> &arr) {
    for(int i=0; i<arr.size(); i++) {
        cin>>arr[i];
    }
}
//Approach Rohit
void displayarrreverse(vector<int> &arr, int n) {
    if(n==-1) {
        return;
    }
    cout<<arr[n]<<endl;
    displayarrreverse(arr,n-1);
}

int main() {
    int n;
    cin>>n;
    vector<int>arr(n,0);
    input(arr);
    displayarrreverse(arr,n-1);

    return 0;
}